kpconv_torch/datasets/S3DIS.py

```diff
@@ -865,20 +865,34 @@ def load_evaluation_points(self, file_path):
     def read_input(self, filename):
         """Read all the input files that belong to the dataset
 
-        Ply files are read by training and testing commands.
+        PLY files are read by training and testing commands.
         """
-        file_extension = filename.split(".")[-1]
-        if file_extension == "ply":
+        file_extension = Path(filename).suffix
+        if file_extension == ".ply":
             # Read ply file
             data = read_ply(filename)
             points = np.vstack((data["x"], data["y"], data["z"])).T
             colors = np.vstack((data["red"], data["green"], data["blue"])).T
-            labels = data["classification"]
-        elif file_extension == "xyz":
+            labels = data["class"]
+        elif file_extension == ".xyz":
             data = np.loadtxt(filename, delimiter=" ")
             points = data[:, :3].astype(np.float32)
             colors = data[:, 3:].astype(np.uint8)
             labels = np.zeros(shape=(data.shape[0],)).astype(np.int32)
+        elif file_extension == ".las" or file_extension == ".laz":
```

Comment:
I'm wondering if some refactoring could be relevant there. We are doing a lot of things in this function, and dispatching the code in accurate module is preferable.

Something like:

```python
if file_extension == ".ply":
    points, colors, labels = read_ply(filename)
elif file_extension == ".xyz":
    points, colors, labels = read_xyz(filename)
elif file_extension in (".las", ".laz"):
    points, colors, labels = read_las(filename)
else:
     # raise error
```

Reply:
ok, done